hydrolib/core/io/bc/models.py

```diff
@@ -220,23 +271,177 @@ class T3D(ForcingBase):
+        "vertpositiontype", enum=VerticalPositionType
+    )
+    _timeinterpolation_validator = get_enum_validator(
+        "timeinterpolation", enum=TimeInterpolation
     )
 
+    @root_validator(pre=True)
+    def _validate_quantityunitpairs(cls, values: Dict) -> Dict:
+        super()._validate_quantityunitpair(values)
+
+        quantityunitpairs = values["quantityunitpair"]
+
+        T3D._validate_that_first_unit_is_time_and_has_no_verticalposition(
+            quantityunitpairs
+        )
+
+        verticalpositions = values.get("vertpositions")
+        if verticalpositions is None:
+            raise ValueError("vertpositions is not provided")
+        verticalpositionindexes = values.get("vertpositionindex")
+
+        number_of_verticalpositions = (
+            len(verticalpositions)
+            if isinstance(verticalpositions, List)
+            else len(verticalpositions.split())
+        )
+
+        if verticalpositionindexes is None:
+            T3D._validate_that_all_non_time_quantityunitpairs_have_valid_verticalpositionindex(
+                quantityunitpairs, number_of_verticalpositions
+            )
+            return values
+
+        T3D._validate_verticalpositionindexes_and_update_quantityunitpairs(
+            verticalpositionindexes,
+            number_of_verticalpositions,
+            quantityunitpairs,
+        )
+
+        return values
+
+    @staticmethod
+    def _validate_that_first_unit_is_time_and_has_no_verticalposition(
+        quantityunitpairs: List[QuantityUnitPair],
+    ) -> None:
+        if quantityunitpairs[0].quantity.lower() != "time":
+            raise ValueError("First quantity should be `time`")
+        if quantityunitpairs[0].vertpositionindex is not None:
+            raise ValueError("`time` quantity cannot have vertical position index")
+
+    @staticmethod
+    def _validate_that_all_non_time_quantityunitpairs_have_valid_verticalpositionindex(
+        quantityunitpairs: List[QuantityUnitPair], maximum_verticalpositionindex: int
+    ) -> None:
+        for quantityunitpair in quantityunitpairs:
+            quantity = quantityunitpair.quantity.lower()
+            verticalpositionindex = quantityunitpair.vertpositionindex
+
+            if quantity == "time":
+                continue
+            if not T3D._is_valid_verticalpositionindex(
+                verticalpositionindex, maximum_verticalpositionindex
+            ):
+                raise ValueError(
+                    f"Vertical position index should be between 1 and {maximum_verticalpositionindex}, but {verticalpositionindex} was given"
+                )
+
+    @staticmethod
+    def _validate_verticalpositionindexes_and_update_quantityunitpairs(
+        verticalpositionindexes: List[int],
+        number_of_verticalpositions: int,
+        quantityunitpairs: List[QuantityUnitPair],
+    ) -> None:
+        if verticalpositionindexes is None:
+            raise ValueError("vertpositionindex is not provided")
```

Comment:
```suggestion
            raise ValueError("vertPositionIndex is not provided")
```

Reply:
@priscavdsluis : I'm fine with this (I even like it a bit better). Only we are not consistent throughout the code: sometimes we print the class/Pydantic fields names (lowercased), and sometimes the alias. Question is mainly: are we adressing the end-user or a core developer? (I vote: end user)

So, I'll commit this suggested change, but we should do it in more places.